packages/react/src/lib/accessibility/Behaviors/Dialog/dialogBehavior.ts

```diff
@@ -39,41 +39,33 @@ const dialogBehavior: Accessibility<DialogBehaviorProps> = props => {
 }
 
 /**
- * Returns the element id of the header or generates a default one. It is
- * used when user does not provide aria-label or aria-labelledby as
- * props. It is also used as default value for header id if there is not
- * any value provided by user as prop.
+ * Returns the element id of the header, it is used when user does not provide aria-label or
+ * aria-labelledby as props.
  */
-const getDefaultAriaLabelledBy = (props: any) => {
-  const { header } = props
-  if (props['aria-label'] || props['aria-labelledby'] || !header) {
+const getDefaultAriaLabelledBy = (props: DialogBehaviorProps) => {
+  if (props['aria-label'] || props['aria-labelledby'] || _.isNil(props.header)) {
     return undefined
   }
-  return header['id'] || _.uniqueId('dialog-header-')
+  return props.headerId
 }
 
 /**
- * Returns the element id of the content or generates a default one. It is
- * used when user does not provide aria-describedby as props. It is also
- * used as default value for content id if there is not any value provided by
- * user as prop.
+ * Returns the element id of the content, it is used when user does not provide aria-describedby
+ * as props.
  */
-const getDefaultAriaDescribedBy = (props: any) => {
-  const { content } = props
-  if (props['aria-describedby'] || !content) {
+const getDefaultAriaDescribedBy = (props: DialogBehaviorProps) => {
+  if (props['aria-describedby'] || _.isNil(props.content)) {
     return undefined
   }
-  return content['id'] || _.uniqueId('dialog-content-')
+  return props.contentId
 }
 
 export default dialogBehavior
 
 type DialogBehaviorProps = {
-  header?: {
-    id?: string
-  }
-  content?: {
-    id?: string
-  }
+  header?: string | object
```

Comment:
are `header` and `content` props used anywhere in the behavior's logic? I mean, even from semantical point of view as a client I wouldn't expect them to be necessary to calculate accessibility attributes - am I missing something?

Reply:
`header` & `content` are used in `getDefaultAriaLabelledBy` & `getDefaultAriaDescribedBy` logic there is quite complicated.

I will introduce `hasContent` & `hasHeader` instead, it has more sense.